shared/actions/chat.js

```diff
@@ -230,23 +231,24 @@ function * _postMessage (action: PostMessage): SagaGenerator<any, any> {
 }
 
 function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
-  switch (action.payload.activity.activityType) {
+  const activity = action.payload.activity.thingValue()
+  switch (activity.activityType) {
     case NotifyChatChatActivityType.incomingMessage:
-      const incomingMessage: ?IncomingMessageRPCType = action.payload.activity.incomingMessage
+      const incomingMessage: ?IncomingMessageRPCType = activity.incomingMessage
       if (incomingMessage) {
         const messageUnboxed: MessageUnboxed = incomingMessage.message
         const yourName = yield select(usernameSelector)
         const conversationIDKey = conversationIDToKey(incomingMessage.convID)
         const message = _unboxedToMessage(messageUnboxed, 0, yourName, conversationIDKey)
 
         // TODO short-term if we haven't seen this in the conversation list we'll refresh the inbox. Instead do an integration w/ gregor
-        const conversationStateSelector = (state: TypedState) => state.chat.get('conversationStates', Map()).get(conversationIDKey)
+        const conversationStateSelector: (state: TypedState) => string = (state: TypedState) => state.chat.get('conversationStates')
```

Comment:
I'm confused how this type checks... shouldn't `state.chat.get('conversationStates')` be a `Map`?